lib/node_modules/@stdlib/_tools/remark/plugins/remark-lint-html-section-structure/lib/main.js

```diff
@@ -0,0 +1,300 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var logger = require( 'debug' );
+var rule = require( 'unified-lint-rule' );
+var visit = require( 'unist-util-visit' );
+
+
+// VARIABLES //
+
+var debug = logger( 'remark-lint-html-section-structure' );
+var SECTION_START = /<section(?:\s+class="([^"]*)")?>/;
+var SECTION_END = /<\/section>$/;
+var SECTION_END_WITH_COMMENT = /<\/section>\s*<!--\s*\/\.([^*]*)\s*-->/;
+var END_COMMENT = /^\s*<!--\s*\/\.([^\s]*)\s*-->/;
+var END_COMMENT_SIMPLE = /<!--\s*\/\.([^>]+)-->/;
+var htmlSectionStructureRule;
+
+
+// MAIN //
+
+/**
+* Validates HTML section structure in README files.
+*
+* @private
+* @param {Node} tree - abstract syntax tree (AST)
+* @param {File} file - virtual file
+* @param {Object} options - options
+* @param {Callback} clbk - callback to invoke upon completion
+* @returns {void}
+*/
+function linter( tree, file, options, clbk ) {
+	var isValidWhitespace;
```

Comment:
Why are all these variables declared in this scope and not in the function scope in which they are used?

Reply:
Addressed.